include/picongpu/particles/collision/InterCollision.hpp

```diff
@@ -201,23 +201,15 @@ namespace picongpu
                             (*longParList).shuffle(worker, rngHandle);
                         });
 
-                    auto collisionFunctorCtx = lockstep::makeVar<decltype(collisionFunctor(
-                        worker,
-                        alpaka::core::declval<DataSpace<simDim> const>(),
-                        alpaka::core::declval<float_X const>(),
-                        alpaka::core::declval<float_X const>(),
-                        alpaka::core::declval<uint32_t const>()))>(forEachFrameElem);
-
-                    forEachFrameElem(
-                        [&](lockstep::Idx const idx)
+                    auto collisionFunctorCtx = forEachFrameElem(
```

Comment:
@pordyna Maybe the change in these lines is of interest to you. There is no need anymore to know the signature of the functor before the creation.